isthmus/src/main/java/io/substrait/isthmus/expression/ExpressionRexConverter.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.util.TimeString;
+import org.apache.calcite.util.TimestampString;
+
+public class ExpressionRexConverter extends AbstractExpressionVisitor<RexNode, RuntimeException> {
+  private final RelDataTypeFactory typeFactory;
+  private final RexBuilder rexBuilder;
+  private final ScalarFunctionConverter scalarFunctionConverter;
+
+  public ExpressionRexConverter(
+      RelDataTypeFactory typeFactory, ScalarFunctionConverter scalarFunctionConverter) {
+    this.typeFactory = typeFactory;
+    this.rexBuilder = new RexBuilder(typeFactory);
+    this.scalarFunctionConverter = scalarFunctionConverter;
+  }
+
+  @Override
+  public RexNode visit(Expression.NullLiteral expr) throws RuntimeException {
+    return null;
+  }
+
+  @Override
+  public RexNode visit(Expression.BoolLiteral expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value());
+  }
+
+  @Override
+  public RexNode visit(Expression.I8Literal expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.I16Literal expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.I32Literal expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.I64Literal expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.FP32Literal expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.FP64Literal expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.StrLiteral expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.BinaryLiteral expr) throws RuntimeException {
+    // Calcite RexLiteral only takes ByteString
+    return rexBuilder.makeLiteral(
+        new ByteString(expr.value().toByteArray()),
+        TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.TimeLiteral expr) throws RuntimeException {
+    // Expression.TimeLiteral is Micros, while RexLiteral assumes Milliseconds for time type
```

Comment:
This comment and the one below are misleading. It sounds like we're losing resolution but we aren't.

Reply:
The comment is not right. I changed the comment.